docker/docker_image_src.go

```diff
@@ -65,11 +73,37 @@ func (s *dockerImageSource) GetManifest() (manifest types.ImageManifest, unverif
 	if res.StatusCode != http.StatusOK {
 		return nil, "", errFetchManifest{res.StatusCode, manblob}
 	}
+	// TODO(remove, already set in manifest, below): Miloslav?
 	unverifiedCanonicalDigest = res.Header.Get("Docker-Content-Digest")
-	manifest, err = makeImageManifest(s.ref.FullName(), manblob, unverifiedCanonicalDigest, nil)
+	tags, err := s.getTags()
+	if err != nil {
+		return nil, "", err
+	}
+	manifest, err = makeImageManifest(s.ref.FullName(), manblob, res.Header.Get("Docker-Content-Digest"), tags)
 	return
 }
 
+func (s *dockerImageSource) getTags() ([]string, error) {
+	url := fmt.Sprintf(tagsURL, s.ref.RemoteName())
+	res, err := s.c.makeRequest("GET", url, nil, nil)
```

Comment:
I don’t understand this. AFAICT the list of tags in a repository (`rhel7/rhel` → `7.0`, `7.0.1`, `7.2`, …) has nothing to do with the single digest or tag reference to which this `dockerImageSource` points (`rhel7/rhel:latest`). Why is the list of all tags a property of the image, and why does `skopeo inspect` of a single image list all of the unrelated tags?

It may well make sense for `skopeo inspect` to provide the information, but the `ImageSource` abstraction is not suited for that; that would be a better fit for an `ImageRepository` of some kind (which in turn could not be used to talk about `save`/`load` streams transfered from stdin/to stdout).

Perhaps a cleaner thing to do would be to lift this all the way to `inspect.go`, and give `inspect.go` an API to get the underlying `dockerClient`? I don’t know, I don’t feel like I understand the requirements.